applications/mne_analyze/plugins/forwardsolution/forwardsolution.cpp

```diff
@@ -0,0 +1,392 @@
+ *     * Neither the name of MNE-CPP authors nor the names of its contributors may be used
+ *       to endorse or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * @brief    ForwardSolution class defintion.
+ *
+ */
+
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "forwardsolution.h"
+
+#include <anShared/Management/communicator.h>
+#include <anShared/Management/analyzedata.h>
+
+#include <anShared/Model/abstractmodel.h>
+#include <anShared/Model/forwardsolutionmodel.h>
+#include <anShared/Model/fiffrawviewmodel.h>
+#include <anShared/Model/averagingdatamodel.h>
+
+#include <disp/viewers/fwdsettingsview.h>
+
+#include <fiff/fiff_stream.h>
+
+#include <fwd/computeFwd/compute_fwd.h>
+#include <fwd/computeFwd/compute_fwd_settings.h>
+
+#include <fs/annotationset.h>
+
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QDebug>
+#include <QDockWidget>
+#include <QVBoxLayout>
+
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace FORWARDSOLUTIONPLUGIN;
+using namespace ANSHAREDLIB;
+
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+ForwardSolution::ForwardSolution()
+: m_pCommu(Q_NULLPTR)
+, m_pFwdSettings(new FWDLIB::ComputeFwdSettings)
+, m_bBusy(false)
+, m_bDoRecomputation(false)
+, m_bDoClustering(true)
+, m_bDoFwdComputation(false)
```

Comment:
Consider using in-class member initialisation instead of using the default constructor. The following core guideline is related. Your case does not 100% match the guideline example since you actually do something other than initialising members. Still I think it would make it more apparent to the reader of the class what the default values are. 

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c45-dont-define-a-default-constructor-that-only-initializes-data-members-use-in-class-member-initializers-instead